src/foam/nanos/auth/AuthenticatedGroupDAO.java

```diff
@@ -0,0 +1,65 @@
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.util.SafetyUtil;
+
+/**
+ * Custom authentication for groupDAO
+ *
+ * Features:
+ *   - If creating a group, you can only give it permissions you already have.
```

Comment:
"permissions implied by those that you already have"

Which is slightly different.